src/SilverStripeS3ProtectedAdapter.php

```diff
@@ -30,19 +47,80 @@ public function findBucketPrefix()
         if (getenv('AWS_PROTECTED_BUCKET_PREFIX') !== false) {
             $prefix = (string) getenv('AWS_PROTECTED_BUCKET_PREFIX');
         }
+
         return $prefix;
     }
 
     /**
+     * @return string
+     */
+    public function findCloudFrontPrivateKeyId()
+    {
+        if (getenv('AWS_CLOUDFRONT_PRIVATE_KEY_ID') !== false) {
+            return (string) getenv('AWS_CLOUDFRONT_PRIVATE_KEY_ID');
+        }
+        throw new LogicException('AWS_CLOUDFRONT_PRIVATE_KEY_ID environment variable not set');
+    }
+
+    /**
+     * @return string
+     */
+    public function findCloudFrontPrivateKeyPath()
+    {
+        if (getenv('AWS_CLOUDFRONT_PRIVATE_KEY_PATH') !== false) {
+            return (string) getenv('AWS_CLOUDFRONT_PRIVATE_KEY_PATH');
+        }
+        throw new LogicException('AWS_CLOUDFRONT_PRIVATE_KEY_PATH environment variable not set');
```

Comment:
Could we make it fall back to S3 if the private key is not found? It's much easier to delete the key from servers + vault instead of removing variables